src/random.cpp

```diff
@@ -143,6 +143,34 @@ static bool GetHardwareRand(unsigned char* ent32) noexcept {
     return false;
 }
 
+/** Use repeated SHA512 to strengthen the randomness in seed32, and feed into hasher. */
+static void Strengthen(const unsigned char* seed32, int microseconds, CSHA512& hasher) noexcept
```

Comment:
nit: passing in seed32's length could be less error prone, in that the function would make fewer assumptions about the arguments / have fewer preconditions for them. Could be good to document the size precondition in the comment as an alternative.

Reply:
Fixed by using the C++11 "pass a reference to array of fixed size" construction (which will cause a compilation failure of the array has the wrong length).